keras_nlp/models/bert.py

```diff
@@ -319,6 +341,28 @@ def BertBase(name=None, trainable=True):
         trainable=trainable,
     )
 
-    # TODO(jbischof): add some documentation or magic to load our checkpoints
+    # TODO(jbischof): add tests for basic functionality but not file loading
+    if weights:
+        if weights not in checkpoints["bert_base"]:
+            raise ValueError(
+                f"""`weights` must be one of {", ".join(checkpoints["bert_base"])}"""
+            )
+        filepath = keras.utils.get_file(
+            weights,
+            BASE_PATH + weights + "/model.h5",
+            cache_subdir="models",
+            file_hash=checkpoints["bert_base"][weights]["md5"],
+        )
+        model.load_weights(filepath)
```

Comment:
Is there a reason we don't want to also support users passing in a custom filepath here as well? It would just be a few lines different, and match the cv applications approach.

Reply:
I thought this functionality was confusing and low value. You can always call `model.set_weights()` on the return value, right?

Reply:
I guess I was more thinking why not? It's a few lines of change that would bring us into agreement with Keras/KerasCV. I guess we could add this after this PR, so not a big deal to do it here or not.

But why don't we want to do this generally? Is it worry with vocabulary_size argument? Or a more general thought that the current keras application API is bad here?

Reply:
I think the concern is that it's too confusing to allow `weights` to be our checkpoints names OR a file. In addition, it's very easy for users to load their own checkpoints (or saved models) on the output. Duplicating this functionality seems like an unnecessary step toward monolithic APIs.

Reply:
Hmm, may be worth finding some time to discuss this more. For WordPieceTokenizer `vocabulary` can be a filename or a list. For SentencePieceTokenizer `proto` can be a serialized proto string, or filename string.

Fine to leave as is for now, because this isn't excluding adding `weights` as a file, but we should try to decide collectively if that's an pattern of anti-pattern.

Reply:
If you add a lot of checkpoint/vocab value with your tokenization then I agree this could make sense. For now I assume you might have to change the API and don't want to overengineer it.

Reply:
sgtm re resolving this along with preprocessing!

Reply:
FYI Keras applications and KerasCV accept local filepaths as `weights`. The logic is to check if the name is built-in keyword ("imagenet") and if not try to load as a local path.

Agree that it's low value, but it's cheap and not doing it is inconsistent.

Reply:
Sounds good! Filed #314 to track as we need to finalize a preprocessing API first.